homeassistant/components/home_plus_control/helpers.py

```diff
@@ -0,0 +1,93 @@
+"""Helper classes and functions for the Legrand Home+ Control integration."""
+import logging
+
+from homeassistant.const import CONF_CLIENT_ID, CONF_CLIENT_SECRET
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_entry_oauth2_flow
+
+from .const import CONF_SUBSCRIPTION_KEY, DOMAIN, OAUTH2_AUTHORIZE, OAUTH2_TOKEN
+from .switch import HomeControlSwitchEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_add_entities(new_unique_ids, coordinator, add_entities):
```

Comment:
This doesn't need to be a coroutine function since we don't await inside.
```suggestion
@callback
def async_add_entities(new_unique_ids, coordinator, add_entities):
```

Reply:
I have included the entity class as one of the parameters that is passed through the dispatcher call. This way the helper function can create the expected entity instance dynamically.
However I struggle to see how the partial could be used here and how this can also be tied to the coordinator creation being moved to the `__init__.py` module as per the other change discussed here: https://github.com/home-assistant/core/pull/46783#discussion_r590205607

Would you be able to provide more insight into how this can be done?

Thank you!

Reply:
Something like this. Although since we only have one platform so far, I think the helper may be a premature optimization. we could remove the helper and just define the add entity callback in the platform for now.
```py
add_entities = partial(helper_add_entities, entity_class=SomeEntity)

remove = async_dispatcher_connect(hass, signal, add_entities)
```

Reply:
OK, thanks. 

As you mention, I have just included the callback in the platform for the moment.